dozer-ingestion/src/connectors/postgres/connection/validator.rs

```diff
@@ -176,17 +177,50 @@ fn validate_tables(
         )
         .map_err(InvalidQueryError)?;
 
+    let mut error_columns = String::new();
+
     for r in result.iter() {
         let table_name: String = r.try_get(0).map_err(InvalidQueryError)?;
+
+        let columns = table_info
+            .iter()
+            .find(|x| x.table_name == table_name)
+            .unwrap()
+            .clone()
+            .columns;
+
+        if let Some(column_info) = columns {
+            let column_result = client
+                .query(
+                    "SELECT column_name FROM information_schema.columns WHERE table_name = ($1)",
```

Comment:
It doesn't look efficient to run query on every table separately

Reply:
updated!